meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -1,17 +1,144 @@
 # common boot handlers
 
+# uEnv handling
 setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
 setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_bootenv 'setenv bootfirmware_version; ext4load ${devtype} ${devnum}:2 ${loadaddr} ${ostree_root}/usr/lib/firmware/version.txt; env import -t ${loadaddr} ${filesize}'
+setenv bootcmd_getroot 'setexpr ostree_root gsub "^.*ostree=([^ ]*).*$" "\\\\1" "${bootargs}"'
+
+# Env saving
+setenv saveenv_mmc 'if test -z "${board_is_closed}"; then saveenv; fi;'
+
+# Boot firmware update helpers
+setenv load_image 'ext4load ${devtype} ${devnum}:2 ${loadaddr} ${image_path}'
+setenv set_blkcnt 'setexpr blkcnt ${filesize} + 0x1ff && setexpr blkcnt ${blkcnt} / 0x200'
+setenv update_image 'run set_blkcnt && mmc dev ${devnum} ${uboot_hwpart} && mmc write ${loadaddr} ${start_blk} ${blkcnt}'
+
+# Import uEnv.txt
+run bootcmd_otenv
+# Get deployment sysroot absolute path
+run bootcmd_getroot
+# Check state of SECONDARY_BOOT bit
+imx_secondary_boot
+# Check if board is on closed state
+imx_is_closed
+
+
+# Device what storage we should use for update-related vars
+# If board is closed - RPMB on eMMC
+# If boars is open - uboot environment
+if test -n "${board_is_closed}"; then
+	echo "Using fiovb"
+else
+	echo "Using ubootenv"
+	setenv fiovb.bootcount "${bootcount}"
+	setenv fiovb.rollback "${rollback}"
+	setenv fiovb.upgrade_available "${upgrade_available}"
+	setenv fiovb.bootupgrade_available "${bootupgrade_available}"
+fi
+
+
+# Handle boot firmware updates
+if test "${fiovb.is_secondary_boot}" = "0"; then
+	if test -n "${board_is_closed}"; then
+		# FIO bootcount, upgrade_available and rollback initialization
+		mmc rescan
+		fiovb init ${devnum}
+
+		fiovb read_pvalue bootcount 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootcount 0; fi
+		fiovb read_pvalue rollback 4
+		if test ! $? -eq 0; then fiovb write_pvalue rollback 0; fi
+		fiovb read_pvalue upgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue upgrade_available 0; fi
+		fiovb read_pvalue bootupgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootupgrade_available 0; fi
+	fi
+
+	# Only update bootcount when upgrade_available is set
+	if test "${fiovb.upgrade_available}" = "1"; then
+		setexpr bootcount ${fiovb.bootcount} + 1;
+		if test -n "${board_is_closed}"; then
+			fiovb write_pvalue bootcount ${bootcount};
+		fi;
+
+		# If bootcount is above bootlimit, trigger rollback
+		# if not - try to boot secondary boot path for boot firmware validation
+		if test ${bootcount} -gt ${bootlimit}; then
+			echo "FIO: doing rollback..."
+			if test -n "${board_is_closed}"; then
+				fiovb write_pvalue rollback 1;
+				fiovb write_pvalue upgrade_available 0;
+				fiovb write_pvalue bootupgrade_available 0;
+			else
+				setenv rollback 1;
```

Comment:
Shouldn't this be fiovb.rollback? Just because you're checking for "${fiovb.rollback}" at bootcmd_rollback.

Reply:
I think we should set both, in the case if user decided to reset device just after execution of this command (so on next boot we will continue rollback procedure)